client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java

```diff
@@ -302,7 +306,18 @@ public void testDeleteJob_GivenWaitForCompletionIsFalse() throws Exception {
         DeleteJobResponse response = execute(deleteJobRequest, machineLearningClient::deleteJob, machineLearningClient::deleteJobAsync);
 
         assertNull(response.getAcknowledged());
-        assertNotNull(response.getTask());
+
+        final TaskId taskId = response.getTask();
```

Comment:
One thing I don't fully understand is, how is this not caught by us ensure that no tasks are running after a test? Shouldn't this task show up in `_cat/tasks` and the wait we do here run there as a result?

Reply:
The way the test works (before this PR) basically fire a job deletion and forgets about it. The delete job action is quite complicated and involves multiple steps, as well as storing the task result, so I suspect that at some point the task manager has no in-flight active tasks and allows to move to the next tests but the store result indexing task is not yet triggered (an registered) and is executed much later.

Reply:
(the test failure can be reproduced by executing the two tests one after the other)

Reply:
> The way the test works (before this PR) basically fire a job deletion and forgets about it. The delete job action is quite complicated and involves multiple steps, as well as storing the task result, so I suspect that at some point the task manager has no in-flight active tasks and allows to move to the next tests but the store result indexing task is not yet triggered (an registered) and is executed much later.

Got it, that makes sense, thanks! I wonder if the temporary absence of any task should be interpreted as a bug ... regardless LGTM :)